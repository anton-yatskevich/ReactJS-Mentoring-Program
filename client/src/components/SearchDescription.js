import React from 'react';
import PropTypes from 'prop-types';

const SearchDescription = ({ numberOfResults, sortField, onChangeSort }) => {
    const sortTypes = [{ value: 'release date', key: 1 }, { value: 'rating', key: 2 }];

    return (
        <div className="search-description__wrapper">
            <div className="search-description">
                <p className="search-description__info">
                    {`${numberOfResults} movies found`}
                </p>
                <div>
                    <span>Sort by:</span>
                    {
                        sortTypes.map(sortType => (
                            <button
                                type="button"
                                key={sortType.key}
                                onClick={() => onChangeSort(sortType.value)}
                                className={sortType.value === sortField ? 'sort-type--active' : ''}
                            >
                                {sortType.value}
                            </button>
                        ))
                    }
                </div>
            </div>
        </div>
    );
};

SearchDescription.propTypes = {
    numberOfResults: PropTypes.number,
    sortField: PropTypes.string,
    onChangeSort: PropTypes.func
};

SearchDescription.defaultProps = {
    numberOfResults: 0,
    sortField: '',
    onChangeSort: null
};

export default SearchDescription;
