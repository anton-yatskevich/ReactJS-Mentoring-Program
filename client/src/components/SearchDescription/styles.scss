.search-description {
	display: flex;
	justify-content: space-between;
	align-items: center;
	max-width: 1200px;
	margin: 0 auto;
	font-weight: 600;

	&__wrapper {
		width: 100%;
		background-color: rgb(209, 209, 209);
		padding: 10px 20px;
	}
}