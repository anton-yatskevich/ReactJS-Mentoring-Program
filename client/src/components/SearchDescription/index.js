import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash.noop';

import SortTypeSelect from '../SortTypeSelect';
import './styles.scss';

const SearchDescription = ({ numberOfResults, sortField, onChangeSort }) => (
    <div className="search-description__wrapper">
        <div className="search-description">
            <p className="search-description__info">{`${numberOfResults} movies found`}</p>
            <SortTypeSelect sortField={sortField} onChangeSort={onChangeSort} />
        </div>
    </div>
);

SearchDescription.propTypes = {
    numberOfResults: PropTypes.number,
    sortField: PropTypes.string,
    onChangeSort: PropTypes.func
};

SearchDescription.defaultProps = {
    numberOfResults: 0,
    sortField: '',
    onChangeSort: noop
};

export default SearchDescription;
