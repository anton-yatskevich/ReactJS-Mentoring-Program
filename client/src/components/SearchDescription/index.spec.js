import React from 'react';
import { shallow } from 'enzyme';
import SearchDescription from './index';

describe('SearchDescription component', () => {
    it('should be rendered with default props', () => {
        const SearchDescriptionComponent = shallow(<SearchDescription />);
        expect(SearchDescriptionComponent).toMatchSnapshot();
    });
});
