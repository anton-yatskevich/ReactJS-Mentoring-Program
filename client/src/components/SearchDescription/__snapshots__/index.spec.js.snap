// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDescription component should be rendered with default props 1`] = `
<div
  className="search-description__wrapper"
>
  <div
    className="search-description"
  >
    <p
      className="search-description__info"
    >
      0 movies found
    </p>
    <SortTypeSelect
      onChangeSort={[Function]}
      sortField=""
    />
  </div>
</div>
`;
